mollie_subscription_ept/controllers/main.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+import logging
+import werkzeug
+from odoo import fields, http
+from odoo.http import request, Response
+from odoo.tools.json import scriptsafe as json_scriptsafe
+
+_logger = logging.getLogger(__name__)
+
+
+class WebsiteSale(http.Controller):
+
+    @http.route(['/clear_warn'], type='json', auth="public", website=True)
+    def clear_warning(self, **post):
+        hide_warn = post.get('hide_warn')
+        if hide_warn:
+            request.session['show_wishlist_warn'] = request.session['show_in_cart_warn'] = False
+        return True
+
+    def _set_session(self, so_obj):
+        request.session['show_wishlist_warn'] = True if so_obj.get('show_wishlist_warning') else False
+        request.session['show_in_cart_warn'] = True if so_obj.get('show_in_cart_warning') else False
+
+    @http.route(['/shop/cart/update'], type='http', auth="public", methods=['POST'], website=True)
+    def cart_update(self, product_id, add_qty=1, set_qty=0, **kw):
+        """This route is called when adding a product to cart (no options)."""
+        sale_order = request.website.sale_get_order(force_create=True)
+        if sale_order.state != 'draft':
+            request.session['sale_order_id'] = None
+            sale_order = request.website.sale_get_order(force_create=True)
+
+        product_custom_attribute_values = None
+        if kw.get('product_custom_attribute_values'):
+            product_custom_attribute_values = json_scriptsafe.loads(kw.get('product_custom_attribute_values'))
+
+        no_variant_attribute_values = None
+        if kw.get('no_variant_attribute_values'):
+            no_variant_attribute_values = json_scriptsafe.loads(kw.get('no_variant_attribute_values'))
+
+        values = sale_order._cart_update(
+            product_id=int(product_id),
+            add_qty=add_qty,
+            set_qty=set_qty,
+            product_custom_attribute_values=product_custom_attribute_values,
+            no_variant_attribute_values=no_variant_attribute_values)
+
+        self._set_session(values)
+
+        if kw.get('express'):
+            return request.redirect("/shop/checkout?express=1")
+
+        return request.redirect("/shop/cart")
+
+    @http.route(['/shop/cart/update_json'], type='json', auth="public", methods=['POST'], website=True, csrf=False)
+    def cart_update_json(self, product_id, line_id=None, add_qty=None, set_qty=None, display=True, **kw):
```

Comment:
The function should not be overriden, make use of super call, else it may affect the working of other modules